programs/validator-history/src/state.rs

```diff
@@ -369,10 +393,24 @@ impl ValidatorHistory {
 
     pub fn set_commission_and_slot(&mut self, epoch: u16, commission: u8, slot: u64) -> Result<()> {
         if let Some(entry) = self.history.last_mut() {
-            if entry.epoch == epoch {
-                entry.commission = commission;
-                entry.vote_account_last_update_slot = slot;
-                return Ok(());
+            match entry.epoch.cmp(&epoch) {
+                Ordering::Equal => {
+                    entry.commission = commission;
+                    entry.vote_account_last_update_slot = slot;
+                    return Ok(());
+                }
+                Ordering::Greater => {
+                    self.history
+                        .arr_mut()
+                        .iter_mut()
+                        .filter(|entry| entry.epoch == epoch)
+                        .for_each(|entry| {
```

Comment:
this implies there may be multiple entries with the same epoch in the CircBuf, which shouldn't be possible given how we're inserting new entries - would prefer to use `.find(|entry| entry.epoch == epoch)` here instead of `.filter().for_each(..)`

Reply:
makes sense